modules/social_features/social_post/src/Hooks/SocialPostFormHooks.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\social_post\Hooks;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hux\Attribute\Alter;
+use Drupal\social_group\CurrentGroupService;
+use Drupal\social_post\Service\SocialPostHelperInterface;
+
+
+final class SocialPostFormHooks {
+
+  private SocialPostHelperInterface $socialPostHelper;
+  private AccountProxyInterface $currentUser;
+
+  public function __construct(
+    SocialPostHelperInterface $social_post_helper,
+    AccountProxyInterface $current_user,
+  ) {
+    $this->socialPostHelper = $social_post_helper;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * From alter hook: replacement of social_post_form_post_form_alter.
+   *
+   * The method definition contains all standard parameters defined by the
+   * definition of the hook.
+   *
+   * @param array $form
+   * @param FormStateInterface $form_state
+   *
+   * @return array
+   */
+  #[Alter('form_post_form')]
+  public function formPostFormAlter(array &$form, FormStateInterface $form_state): void {
+    $content = $this->socialPostHelper->buildCurrentUserImage();
+
+    if ($form_state->getFormObject()->getEntity()->isNew()
+      && $content !== NULL) {
+      $form['current_user_image'] = $content;
+    }
+
+    // Reset title display.
+    $form['field_post']['widget'][0]['#title_display'] = '';
+
+    // Set submit button caption to Post instead of Save.
+    $form['actions']['submit']['#value'] = t('Post', [], ['context' => 'Post button']);
+
+    if (empty($form['field_post']) || empty($form['field_post']['widget'][0])) {
```

Comment:
Clean Code Principles: return early

Reply:
With where this if statement is positioned this will never be true anymore due to `$form['field_post']['widget'][0]['#title_display']`. Guard clauses should usually be at the top.